core/sql/executor/ExHdfsScan.cpp

```diff
@@ -1427,24 +1424,28 @@ char * ExHdfsScanTcb::extractAndTransformAsciiSourceToSqlRow(int &err,
       else
         attr = NULL;
  
-      if (!isTrailingMissingColumn)
-        sourceColEnd = strchr(sourceData, delimiter);
-
-      if(!isTrailingMissingColumn)
-	{
-          short len = 0;
-	  if (sourceColEnd  &&
-	      sourceColEnd <= sourceDataEnd)
-	    len = sourceColEnd - sourceData;
-	  else
-            { 
-              len = sourceDataEnd - sourceData;
-              if (i != hdfsScanTdb().convertSkipListSize_ - 1)
-                isTrailingMissingColumn = TRUE;
-            }
-
-          if (attr) // this is a needed column. We need to convert
-          {
+      if (!isTrailingMissingColumn) {
+         sourceColEnd = hdfs_strchr(sourceData, rd, cd, sourceDataEnd, &rdSeen);
+         if (sourceColEnd == NULL) {
+            if (rdSeen || (sourceRowEnd == NULL))
+               return NULL;
+            else
+               return sourceRowEnd+1;
+         }
+         short len = 0;
+	 len = sourceColEnd - sourceData;
+         if (rdSeen) {
+            sourceRowEnd = sourceColEnd; 
+            hdfsLoggingRowEnd_  = sourceRowEnd;
+            if ((endOfRequestedRange_) && 
+                (sourceRowEnd >= endOfRequestedRange_))
+                   *(sourceRowEnd +1)= RANGE_DELIMITER;
```

Comment:
Ok, I can fix this without the need for cqd by enabling when to look for range delimiter